File: BetESS_a69303_pg25291 11nov_observer/BetESS/src/betess/Observer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package betess;


public interface Observer {
    public void update(int idEvento,String evento,int resultado);
    public void updateOdd(int idEvento);
    
}

File: Ruby_Observer/observer.rb
=begin interface :Observer do
  def update(id_evento,evento,resultado); end
  def update_odd(id_evento); end
end


####
=end
class Observer
  def initialize(observable)
    observable.add_observer(self)
  end

  def update(counter)
    puts "Count has increased by: #{counter}"
  end
end